### PR TITLE
How to use pre-commit

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [ ] enhancement
- [x] documentation
- [ ] dependencies

## Description

How to use pre-commit before every commit.

### CODE REVIEW

File: PRE-COMMIT.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# How to use pre-commit
```
Comment (RobertKrawitz) [inline PRE-COMMIT.md:1]: This wasn't correct on my laptop.  `pip-install pre-commit` did not install a `pre-commit` command; I had to `dnf install pre-commit` to get the pre-commit command.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -13,8 +13,9 @@ repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.4.0
     hooks:
+#- id: check-yaml # Comment it out because it affects all golden test YAML files.
       - id: check-json
-      - id: check-yaml
       - id: trailing-whitespace
+        exclude: benchmark_runner/grafana/perf/jsonnet/main.libsonnet
```
Comment (RobertKrawitz) [inline .pre-commit-config.yaml:19]: As main.libsonnet is a purely generated file, this makes sense.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -13,8 +13,9 @@ repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.4.0
     hooks:
+#- id: check-yaml # Comment it out because it affects all golden test YAML files.
```
Comment (RobertKrawitz) [inline .pre-commit-config.yaml:16]: I'm not entirely sure about this.  The golden files are generated, but problems with them likely indicate problems with the templates used to generate them.  Do you have an example of where this causes problems?

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
- Changes needed to PRE-COMMIT.md.
- I'm not convinced that disabling this check is the right way to approach this issue; the problem may be that the template files need to be fixed.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ebattat) [reply .pre-commit-config.yaml:16]: See all the errors in the attached file:
[precommit_yamls_errors.txt](https://github.com/user-attachments/files/15620071/precommit_yamls_errors.txt)

Comment (RobertKrawitz) [reply .pre-commit-config.yaml:16]: I'm going to take a look at fixing those.

**Review by RobertKrawitz — COMMENTED** [review]
I found a number of other useful hooks to add.  I'm going to open a new PR for all of that.

**Comment by RobertKrawitz** [comment]
I've opened #844 to fix the check-yaml problems and added a number of new hooks in addition.

**Comment by ebattat** [comment]
Merged in [PR](https://github.com/redhat-performance/benchmark-runner/pull/844)

**Comment by ebattat** [comment]
Merged in [PR](https://github.com/redhat-performance/benchmark-runner/pull/844)

**Pull request closed without merging**